File: music_app/www/songPortal.py
import frappe
from frappe import _

@frappe.whitelist(allow_guest=True)
def get_paginated_songs(page_number=1, page_size=10):
    page_number = int(page_number)
    page_size = int(page_size)
    start = (page_number - 1) * page_size

    songs = frappe.db.get_list('Artist_songs', fields=['name', 'artist', 'song_image'], start=start, limit=page_size)
    total_songs = frappe.db.count('Artist_songs')

    return {
        'songs': songs,
        'total_songs': total_songs,
        'page_number': page_number,
        'page_size': page_size
    }

File: music_app/www/songPortal.js
    let currentPage = 1;
    const pageSize = 10;

    async function loadSongs(page) {
        if (page < 1) return;

        const response = await fetch(`/api/method/music_app.www.songPortal.get_paginated_songs?page_number=${page}&page_size=${pageSize}`);
        const data = await response.json();
        const { songs, total_songs, page_number, page_size } = data.message;
        currentPage = page_number;

        const songsContainer = document.getElementById('songs-container');
        songsContainer.innerHTML = '';

        songs.forEach(song => {
            const songCard = `
                <div class="col-md-6">
                    <div class="songCard w-100 border border-3 p-2 m-2 text-center">
                        <img src="${song.song_image}" alt="Image of ${song.song_name}" class="w-75 img-fluid">
                        <h4 class="m-0 p-0">${song.artist}</h4>
                        <h3 class="m-0 p-0 text-info">${song.name}</h3>
                    </div>
                </div>
            `;
            songsContainer.innerHTML += songCard;
        });

        const totalPages = Math.ceil(total_songs / page_size);
        document.getElementById('page-info').innerText = `Page ${page_number} of ${totalPages}`;
        document.getElementById('prev-page').disabled = page_number === 1;
        document.getElementById('next-page').disabled = page_number === totalPages;
    }
    document.addEventListener('DOMContentLoaded', function() {
        loadSongs(currentPage);
    });